optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -1313,7 +1313,10 @@ def forward(
                 )
             # Only compute necessary logits, and do not upcast them to float if we are not computing the loss
             # TODO: remove the float() operation in v4.46
-            logits = self.lm_head(hidden_states[:, -num_logits_to_keep:, :]).float()
+            if reuse_cache:
```

Comment:
Can you also check if mixtral and mistral needs this code.
Prior to this fix for command: `pytest -s -v tests/transformers/tests/models/ -k "contrastive_generate_dynamic_shapes" 
`, we saw three failures: 
```
FAILED tests/transformers/tests/models/llama/test_modeling_llama.py::LlamaModelTest::test_contrastive_generate_dynamic_shapes - AssertionError: Lists differ: [[47, 95, 92, 35, 83, 82]] != [[47, 95, 92, 35, 66, 8]] 
FAILED tests/transformers/tests/models/mistral/test_modeling_mistral.py::MistralModelTest::test_contrastive_generate_dynamic_shapes - AssertionError: Lists differ: [[55, 43, 83, 75, 15, 60]] != [[55, 43, 83, 19, 15, 94]] 
FAILED tests/transformers/tests/models/mixtral/test_modeling_mixtral.py::MixtralModelTest::test_contrastive_generate_dynamic_shapes - AssertionError: Lists differ: [[41, 73, 42, 57, 65, 26]] != [[41, 73, 42, 17, 89, 3]] 
==================================================================== 3 failed, 7 passed, 1425 deselected, 15 warnings in 19.74s ==
```
They pass if I add the same change to mixtral and mistral:
```
========================================================================= 10 passed, 1425 deselected, 15 warnings in 22.51s =========================================================================
```

Reply:
Added same change to mixtral, mistral, qwen, starcoder and phi, resolved the text generation on these models:
‘mistralai/Mistral-7B-Instruct-v0.2'
’Qwen/Qwen2.5-Coder-1.5B‘
’bigcode/starcoder2-15b‘